client/dashboard/domains/domain-overview/index.tsx

```diff
@@ -0,0 +1,65 @@
+import { useQuery, useSuspenseQuery } from '@tanstack/react-query';
+import { Button } from '@wordpress/components';
+import { __, sprintf } from '@wordpress/i18n';
+import { useMemo } from 'react';
+import { useLocale } from '../../app/locale';
+import { domainQuery } from '../../app/queries/domain';
+import { sitePurchaseQuery } from '../../app/queries/site-purchases';
+import { domainRoute } from '../../app/router';
+import { PageHeader } from '../../components/page-header';
+import PageLayout from '../../components/page-layout';
+import { formatDate } from '../../utils/datetime';
+import { getDomainRenewalUrl, isDomainRenewable } from '../../utils/domain';
+
+export default function DomainOverview() {
+	const locale = useLocale();
+	const { domainName } = domainRoute.useParams();
+	const { data: domain } = useSuspenseQuery( domainQuery( domainName ) );
+	const { data: purchase } = useQuery(
+		sitePurchaseQuery( domain.blog_id, parseInt( domain.subscription_id, 10 ) )
+	);
+
+	const wrappableDomainName = useMemo( () => {
+		const [ domainName, ...tlds ] = domain.domain.split( '.' );
+		const tld = tlds.join( '.' );
+
+		return (
+			<span style={ { wordBreak: 'break-word', hyphens: 'none' } }>
+				{ domainName }
+				<span style={ { whiteSpace: 'nowrap' } }>.{ tld }</span>
+			</span>
+		);
+	}, [ domain.domain ] );
+
+	return (
+		<PageLayout
+			size="small"
+			header={
+				<PageHeader
+					title={ wrappableDomainName }
+					// translators: date is the date the domain was registered.
+					description={ sprintf( __( 'Registered on %(date)s' ), {
+						date: formatDate( new Date( domain.registration_date ), locale, { dateStyle: 'long' } ),
+					} ) }
+					actions={
+						isDomainRenewable( domain ) &&
+						purchase && (
+							<Button
+								variant="primary"
+								__next40pxDefaultSize
+								href={ getDomainRenewalUrl( domain, purchase ) }
+							>
+								{
+									// translators: price is the price of the domain renewal.
+									sprintf( __( 'Renew now for %(price)s' ), {
+										price: purchase.price_text,
```

Comment:
@sirbrillig I'm 99% sure we don't want to use this specific variable and instead want to rely on [`purchase-meta-price`](https://github.com/Automattic/wp-calypso/blob/trunk/client/me/purchases/manage-purchase/purchase-meta-price.tsx) here.

If that's the case, I see you've ported other `<Purchase* />` components, so could you please tell me how exactly you did it? Just copied the component over and removed the Calypso references? What would be the ideal place to put this new component?